initial version of akka module

derekmorr opened the pull request:
As discussed on gitter today, here's a module that loads Akka `Timeout` values.

This still fails on 2.10 b/c SBT tries to build for 2.10 (and Akka 2.4 isn't published for 2.10).

derekmorr committed:
initial version of akka module; fails on 2.10

derekmorr committed:
initial version of akka module; fails on 2.10

derekmorr committed:
Merge branch 'master' into akka-module

melrief reviewed (approved):
LGTM, no comments. Now that we have a module for `akka`, I wonder how many other classes we could support.

This is a module and you don't need a second review if you want to. If you agree, I can merge it but we will have to wait for the next version of pureconfig to publish the artifact.

derekmorr commented:
This will fail the builds on Travis bc it doesn't compile on Scala 2.10. I need to figure out how not to run it on 2.10.

derekmorr committed:
update dependencies

derekmorr committed:
Merge branch 'master' into akka-module

derekmorr committed:
update ammonite-repl to 0.8.5

derekmorr committed:
update sbt; scala; scalatest; jodaconvert

derekmorr committed:
update scala versions for travis

remove unused CanBuildFrom parameter

set Xlint based on scalac version; adapted from cats

update example project to 0.7 and scala 2.12

derekmorr committed:
update example project sbt and scalariform

derekmorr committed:
add link to underlying bug for 2.12.2 Xlint

derekmorr committed:
update to scala 2.10.6; switch example to 2.12

ruippeixotog committed:
Merge branch 'master' into akka-module

ruippeixotog committed:
Use sbt-doge to make SBT obey cross-version restrictions

ruippeixotog committed:
Define `timeoutCC` using `xmap`

ruippeixotog committed:
Merge branch 'master' into derekmorr-akka-module

ruippeixotog commented:
@derekmorr I have fixed the SBT build to work in the configuration we want, but I cannot push to this branch as it's in your personal remote. Instead, I pushed the fix to the branch `derekmorr-akka-module` in `pureconfig/pureconfig`. You should be able to pull those commits and do a fast-forward merge to yours. That way we won't lose the history of this PR or any of the commits.

ruippeixotog committed:
Update library orgnization

derekmorr committed:
merge fixes

derekmorr committed:
update akka to 2.4.18

derekmorr committed:
update contributors
